tests/app.test.ts

```diff
@@ -1,8 +1,99 @@
 import request from "supertest";
-import  app from "../source/app";
+import router from '../routes/routes';
+import { Soldier } from "../repository/repo";
+const app = router; 
+beforeEach(async ()=>{ 
+    await Soldier.deleteMany();
+});
 
 describe('Is Soldier Alive', () => {
     it('Should Send an API Get Request If Soldier Is Alive, Expected 200:', () => {
         request(app).get('/health').expect("I am alive!");
     });
 });
+
+describe('Insert Soldier into DB #important: you need to modify the ID of inserted soldier in the test file in order to succeed this test', () => {
+    it('Should Send an API Post Request and save soldier to DB, Expected 201:',async () => {
+
+        const SoldierToBeAdded = {
```

Comment:
Test data and examples should be placed in /test/data/ folder.

Import it from there